### PR TITLE
Wire up admin privilege grant and revoke. Fixes #2872

### LINKED ISSUE #2872: Granting all privileges to a users doesn't work.
The following commands:
```sql
CREATE USER jdoe WITH PASSWORD '1337'
grant all priviledges to jdoe
SHOW USERS
```

Should result in:

```json
{"results":[{"series":[{"columns":["user","admin"],"values":[["jdoe",true]]}]}]}
```

But results in:

```json
{"results":[{"series":[{"columns":["user","admin"],"values":[["jdoe",false]]}]}]}
```

Please see the skipped test in `/cmd/influxed/run/server_test.go` named `TestServer_UserCommands`, table test: `show users, existing user as admin`

### PR DESCRIPTION
**Pull request by gunnaraasen**

### CODE REVIEW

**Comment by gunnaraasen** [comment]
This is ready for review. @otoolep I'm lost on why circle tests are failing.

File: meta/statement_executor.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ type StatementExecutor struct {
 		CreateUser(name, password string, admin bool) (*UserInfo, error)
 		UpdateUser(name, password string) error
 		DropUser(name string) error
-		SetPrivilege(username, database string, p influxql.Privilege) error
+		SetPrivilege(username, database string, p influxql.Privilege, admin bool) error
```
Comment (otoolep) [inline meta/statement_executor.go]: Just wondering -- we couldn't extend the meaning of `influxql.Privilege` to include an admin level, could we?

File: meta/data.go [hunk 2]
```diff
@@ -464,12 +464,17 @@ func (data *Data) UpdateUser(name, hash string) error {
...
 
 // SetPrivilege sets a privilege for a user on a database.
-func (data *Data) SetPrivilege(name, database string, p influxql.Privilege) error {
+func (data *Data) SetPrivilege(name, database string, p influxql.Privilege, admin bool) error {
 	ui := data.User(name)
 	if ui == nil {
 		return ErrUserNotFound
 	}
 
+	if database == "" {
```
Comment (otoolep) [inline meta/data.go]: This needs a comment. It is not clear from the code why if `database == ""` then `ui.Admin` is set. The concepts seems unrelated.

**Comment by otoolep** [comment]
Kicked the build the couple more times, one reason for the CI failures were build issues, which is rather strange.

File: meta/statement_executor.go (hunk 1 above)
Comment (gunnaraasen) [inline meta/statement_executor.go]: I'll open a new PR to see if extending `influxql.Privilege` is a better way to pass along the admin privilege (I suspect it is).

Adding the admin flag seemed cleaner when I initially started this PR, but it means the only way to tell if the admin privilege should be set is by checking if the database is empty.

Comment (otoolep) [inline meta/statement_executor.go]: You mean that the admin flag should be checked if no database is specified? What if a database is specified and the user has admin privileges?

**Comment by gunnaraasen** [comment]
Closing in favor of #3244
